### ISSUE TITLE
M3: Unsubscribe link with email campaign is broken

### ISSUE THREAD

**Issue by dichvuhuuich** [mautic/mautic#8329]
[//]: # ( Invisible comment: 
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Before you create the issue:
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Search for similar report among other reported issues.
Learn how to troubleshoot at https://www.mautic.org/docs/en/tips/troubleshooting.html
Use drag&drop to attach images or other files )

## Bug Description
When click Link unsbscribe in email send by a campaign (generated by token {unsubscribe_text}) go to error page

| Q   | A
| --- | ---
| Mautic version |  3.0.0-alpha
| PHP version |  7.2.26
| Browser |  Chrome Version 79.0.3945.117 (Official Build) (64-bit)

### Steps to reproduce
1. Create email with  token {unsubscribe_text}
2. Create a segment test to send test email with a contact
3. Create a campaign send email with segment test
4. Contact receive mail with unsubscribe link, click go to error page
5. Though unsubscribe is broken but contact still changed to Do Not Contact
 
### Log errors
Uh oh! I think I broke it. If I do it again, please report me to the system administrator! 
500 Internal Server Error
Mautibot
Report an issue

[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

**Comment by dennisameling** [comment]
Can reproduce on fresh Mautic 3 installation. Logs show the following:

```
[2020-01-22 18:43:51] mautic.WARNING: PHP Warning - include(/var/www/vhosts/fits4all.nl/httpdocs/mautic3/vendor/composer/../twig/twig/src/Node/Expression/Test/DefinedTest.php): failed to open stream: No such file or directory - in file /var/www/vhosts/fits4all.nl/httpdocs/mautic3/vendor/composer/ClassLoader.php - at line 444 {"file":"/var/www/vhosts/fits4all.nl/httpdocs/mautic3/vendor/composer/../twig/twig/src/Node/Expression/Test/DefinedTest.php"} []
[2020-01-22 18:43:51] mautic.WARNING: PHP Warning - include(/var/www/vhosts/fits4all.nl/httpdocs/mautic3/vendor/composer/../twig/twig/src/Node/Expression/Test/DefinedTest.php): failed to open stream: No such file or directory - in file /var/www/vhosts/fits4all.nl/httpdocs/mautic3/vendor/composer/ClassLoader.php - at line 444 {"file":"/var/www/vhosts/fits4all.nl/httpdocs/mautic3/vendor/composer/../twig/twig/src/Node/Expression/Test/DefinedTest.php"} []
[2020-01-22 18:43:51] mautic.WARNING: PHP Warning - include(): Failed opening '/var/www/vhosts/fits4all.nl/httpdocs/mautic3/vendor/composer/../twig/twig/src/Node/Expression/Test/DefinedTest.php' for inclusion (include_path='.:/opt/plesk/php/7.2/share/pear') - in file /var/www/vhosts/fits4all.nl/httpdocs/mautic3/vendor/composer/ClassLoader.php - at line 444 {"file":"/var/www/vhosts/fits4all.nl/httpdocs/mautic3/vendor/composer/../twig/twig/src/Node/Expression/Test/DefinedTest.php"} []
[2020-01-22 18:43:58] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\ClassNotFoundException: "Attempted to load class "DefinedTest" from namespace "Twig\Node\Expression\Test". Did you forget a "use" statement for another namespace?" at /var/www/vhosts/fits4all.nl/httpdocs/mautic3/vendor/twig/twig/src/ExpressionParser.php line 693 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\ClassNotFoundException(code: 0): Attempted to load class \"DefinedTest\" from namespace \"Twig\\Node\\Expression\\Test\".\nDid you forget a \"use\" statement for another namespace? at /var/www/vhosts/fits4all.nl/httpdocs/mautic3/vendor/twig/twig/src/ExpressionParser.php:693)"} []
```

**Comment by dennisameling** [comment]
Just did some additional checks. The [Alpha release](https://github.com/mautic/mautic/releases/tag/3.0.0-alpha) doesn't contain the `vendor/twig/twig/src/Node/Expression/Test` folder, while if you checkout the [3.x branch](https://github.com/mautic/mautic/tree/3.x), install the Composer dependencies, and then check, it does contain the `/Test` folder. Then I tried to build Mautic, and in the build package the Test folder is also missing. So I'm pretty sure something's wrong in the build part of Mautic, but would need to dive a bit deeper into it to find out why exactly.

**Comment by LucasBorthiry** [comment]
Hi. I'm running with the same issue.

**Bug description**

> Mautic Version  3.2.5
> PHP version 7.3.31
> Chrome Version 95.0.4638.69 (Build oficial) (x86_64)

**Steps to reproduce**

> Create email with token {unsubscribe_text}
> Create a segment test to send test email with a contact
> Create a campaign send email with segment test
> Contact receive mail with unsubscribe link, click go to error page
> Though unsubscribe is broken but contact still changed to Do Not Contact

**Log errors**

Uh oh! I think I broke it. If I do it again, please report me to the system administrator!
500 Internal Server Error
Mautibot
Report an issue

It's pretty much the same issue. Any thoughts? 

Thanks in advance.

**Comment by escopecz** [comment]
@LucasBorthiry it may be different issue because the one above was fixed in 3.0.0-beta so you should have the fix in your Mautic instance. Please check [Mautic logs](https://docs.mautic.org/en/troubleshooting).
